mono/metadata/custom-attrs.c

```diff
@@ -213,6 +213,30 @@ load_cattr_enum_type (MonoImage *image, const char *p, const char *boundp, const
 	return mono_class_from_mono_type (t);
 }
 
+static MonoString* 
+cattr_string_from_wtf8 (MonoDomain *domain, const char *text, guint length, MonoError *error)
```

Comment:
This is a generic string conversion, it should go into object.c

Reply:
It should be mono_string_new_wtf8_checked

Reply:
I thought only "checked" to discriminate from non-checked.

Reply:
Why does the data have to become utf8 in the first place? We don't store utf16 in the PE and keep it that way?

Reply:
Custom attribute strings are stored as wtf-8 in .NET PE (the spec says utf-8 but .NET framework and coreclr use wtf-8), but our internal string representation is UTF16 so this conversion is necessary at load time

Reply:
I guess they were really optimizing the PE for size given typical English names, but imho this is the wrong level to have done that. Oh well.